src/renderer/start-app.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import "./components/app.scss";
+
+import { bootstrap } from "./bootstrap";
+import type { DiContainer } from "@ogre-tools/injectable";
+import environmentVariablesInjectable from "../common/utils/environment-variables.injectable";
+
+interface AppConfig {
+  di: DiContainer;
+  mode?: "production" | "development";
+}
+
+export function startApp(conf: AppConfig) {
+  const { di, mode } = conf;
+
+  if (mode) {
+    const environmentVariables = di.inject(environmentVariablesInjectable);
+
+    environmentVariables.NODE_ENV = mode;
+  }
```

Comment:
This doesn't seem right. I don't think we should be mutating environment variables in this sense. 

As a temporary solution, how about making `isProduction` and `isDevelopment`  inject `environmentInjectionToken` which is registered here? 

In the end, we should remove `isProduction` and `isDevelopment` for good. Those are used in places where we need some kind of configuration from the client. 

E.g. `createKubeApiForClusterInjectable` uses `isDevelopment` to determinate whether debugging should be enabled or not. 

How about we introduce `kubeApiDebuggingIsEnabledInjectionToken` in the library, but require client to register implementation for it. 

In that way, the client has full control over the configuration and we as a library do not depend from arbitrary environment variables at all.

Reply:
> As a temporary solution, how about making isProduction and isDevelopment inject environmentInjectionToken which is registered here?

I actually had it like that first, it felt also really weird... anyway I can change this.